bundles/org.openhab.binding.ipcamera/src/main/java/org/openhab/binding/ipcamera/internal/DahuaHandler.java

```diff
@@ -48,6 +48,147 @@ public DahuaHandler(IpCameraHandler handler, int nvrChannel) {
+                case "VideoLoss":
+                case "VideoBlind":
+                    if (action.equals("Start")) {
+                        ipCameraHandler.setChannelState(CHANNEL_TOO_DARK_ALARM, OnOffType.ON);
+                    } else if (action.equals("Stop")) {
+                        ipCameraHandler.setChannelState(CHANNEL_TOO_DARK_ALARM, OnOffType.OFF);
+                    }
+                    break;
+                case "VideoAbnormalDetection":
+                    if (action.equals("Start")) {
+                        ipCameraHandler.setChannelState(CHANNEL_SCENE_CHANGE_ALARM, OnOffType.ON);
+                    } else if (action.equals("Stop")) {
+                        ipCameraHandler.setChannelState(CHANNEL_SCENE_CHANGE_ALARM, OnOffType.OFF);
+                    }
+                    break;
+                case "VideoUnFocus":
+                    if (action.equals("Start")) {
+                        ipCameraHandler.setChannelState(CHANNEL_TOO_BLURRY_ALARM, OnOffType.ON);
+                    } else if (action.equals("Stop")) {
+                        ipCameraHandler.setChannelState(CHANNEL_TOO_BLURRY_ALARM, OnOffType.OFF);
+                    }
+                    break;
+                case "AlarmLocal":
+                    if (action.equals("Start")) {
+                        if (content.contains("index=0")) {
+                            ipCameraHandler.setChannelState(CHANNEL_EXTERNAL_ALARM_INPUT, OnOffType.ON);
+                        } else {
+                            ipCameraHandler.setChannelState(CHANNEL_EXTERNAL_ALARM_INPUT2, OnOffType.ON);
+                        }
+                    } else if (action.equals("Stop")) {
+                        if (content.contains("index=0")) {
+                            ipCameraHandler.setChannelState(CHANNEL_EXTERNAL_ALARM_INPUT, OnOffType.OFF);
+                        } else {
+                            ipCameraHandler.setChannelState(CHANNEL_EXTERNAL_ALARM_INPUT2, OnOffType.OFF);
+                        }
+                    }
+                    break;
+                case "LensMaskOpen":
+                    ipCameraHandler.setChannelState(CHANNEL_ENABLE_PRIVACY_MODE, OnOffType.ON);
+                    break;
+                case "LensMaskClose":
+                    ipCameraHandler.setChannelState(CHANNEL_ENABLE_PRIVACY_MODE, OnOffType.OFF);
+                    break;
+                case "TimeChange":
+                case "NTPAdjustTime":
+                case "StorageChange":
+                case "Reboot":
+                case "NewFile":
+                case "VideoMotionInfo":
+                case "RtspSessionDisconnect":
+                case "LeFunctionStatusSync":
+                case "RecordDelete":
+                    break;
+                default:
+                    ipCameraHandler.logger.debug("Unrecognised Dahua event, Code={}, action={}", code, action);
+            }
+        } catch (IndexOutOfBoundsException e) {
```

Comment:
Please check the lengths beforehand. If this can only happen due to a bug in your code, you don't need to catch this unchecked exception as it will be handled by the framework.

Reply:
DONE: Added checks for the lengths and removed the catch which should not occur anymore.